CAR-1 Add Spring Boot Infrastructure

majagabajaga opened the pull request:
Added the following components:
- Spring boot application
- Initial data model (car, service item, user)
- Initial web page (thymeleaf)
- Initial service
- Initial JPA repositories
- default application configuration

Removed app name from settings so that it does not restrict eclipse app
project name.

Added .gitignore

kagbepreved committed:
CAR-1 Add Spring Boot Infrastructure

Added the following components:
- Spring boot application
- Initial data model (car, service item, user)
- Initial web page (thymeleaf)
- Initial service
- Initial JPA repositories
- default application configuration

Removed app name from settings so that it does not restrict eclipse app
project name.

Added .gitignore

kagbepreved committed:
CAR-1 Add Spring Boot Infrastructure

Added /build/ to .gitignore.

mironuk commented on src/main/java/carservice/web/data/model/Car.java:
```diff
@@ -0,0 +1,71 @@
+
+@Entity
+@Table(name="car")
+public class Car {
```
I think Car entity should have user_id column.

mironuk replied:
And this field should have @ManyToOne annotation.

kagbepreved committed:
CAR-1 Add Spring Boot Infrastructure

- Added the first service test
- Added the @Repository annotation to the JPA repositories
- Added the CarServiceTx transactional class.

kagbepreved committed:
CAR-1 Add Spring Boot Infrastructure

Added support for service static web resources.

mironuk commented on src/main/java/carservice/web/data/model/ServiceItem.java:
```diff
@@ -0,0 +1,84 @@
+	private long serviceItemId;
+
+	@Column(name="car_id")
+	private long carId;
```
This field should have @ManyToOne annotation.

kagbepreved committed:
CAR-1 Add Spring Boot Infrastructure

- Added the userId property to the Car class.
- Added the new line character to the files where it was missing.
